addons/mail/models/mail_message.py

```diff
@@ -121,7 +121,9 @@ def _get_default_author(self):
     need_moderation = fields.Boolean('Need moderation', compute='_compute_need_moderation', search='_search_need_moderation')
     #keep notification layout informations to be able to generate mail again
     layout = fields.Char('Layout', copy=False)  # xml id of layout
-    layout_values = fields.Char('Notification values', copy=False)
+    add_sign = fields.Boolean(default=False)
+    mail_auto_delete = fields.Boolean(default=True)
+    model_description = fields.Char()
```

Comment:
This part is more a prototyping and I'll need to discuss about that with tde, but I'll keep this in mind.

Reply:
@tde-banana-odoo Food for thought:

I'm wondering about the semantics of `auto_delete`. It currently drives the auto-deletion of the `mail.mail` entry when it is successfully sent. This, in turn, drives the cascade-deletion of the parent `mail.message`, but only if this was not a notification. Typically, for mass-mailing emails.

I think we should never keep a `mail.mail` entry after it was successfully sent, because that is purely redundant information. The semantics should be shifted to the cascading part only.

And then we need to decide whether we keep failed mail.mail records, or if we delete them. For notifications, we can rely on the mail.notification to store the status and the info to retry, so it would be safe to delete them. But for mass-mailing, if we don't keep the message we have no notification, only a `mail.mail.statistics` record. Is this sufficient?

Reply:
Hello,

Currently there are 3 parameters to check if I am not wrong: auto delete, model description and signature.

About auto_delete: it is used to comply with the auto delete of mail.template. Default behavior is indeed to delete the sent emails as we don't see added value keeping them. When using a mail.template we support the auto delete parameter coming from the template. Current behavior is to try to support most mail.template features in both pure mass mailing and template-based posting processes. I already had people asking me how to keep send emails based on template, meaning this field is used (but not found easily as template configuration is quite complicated, but that's another issue). With the fail notification + resend we could consider people don't have to keep emails sent with a template on documents, meaning we could drop the auto_delete key support. Do you see any use case that could be broken by removing the auto delete support ?

About model description: it is used to be able to regenerate emails for pure notifications. Those are not linked to a document, just a message pushed to some people Inbox/email. As the resend feature is currently limited to documents we don't need it. We don't plan to support the resend for pure notifications as the added values is not obvious. We can therefore remove model description if I am not wrong.

Last thing is the signature addition. This one is a bit tricky as it also comes from the template. Default behavior is to add the author signature in emails when sending notifications. If you use a template that already includes a signature you can uncheck 'add signature' from the template. We have to keep this information to be able to correctly send the notification again. At resend we don't know if the signature should be included or not. It is not included in the message body as in a chatter it is more considered as noisy than interesting. Notification layout therefore requires to know if he has to include it or not. As most messages require the author signature, we could add a column about "not adding signature", meaning this column will be False for most messages. It adds a column on a heavy model but the value will be void by default (and migration will not have to fill it).